lang/src/org/partiql/lang/eval/visitors/StaticTypeInferenceVisitorTransform.kt

```diff
@@ -530,120 +529,74 @@ internal class StaticTypeInferenceVisitorTransform(
             val nAry = super.transformExprCallAgg(node) as PartiqlAst.Expr.CallAgg
             val funcName = nAry.funcName
             // unwrap the type if this is a collectionType
+            // only collection types work for agg functions, e.g. SUM(1) & SUM(NULL) return NULL
             val argType = when (val type = nAry.arg.getStaticType()) {
                 is CollectionType -> type.elementType
                 else -> type
             }
             val sourceLocation = nAry.getStartingSourceLocationMeta()
-            return nAry.withStaticType(computeReturnTypeForAggFunc(funcName, argType, sourceLocation))
+            return nAry.withStaticType(computeReturnTypeForAggFunc(funcName.text, argType, sourceLocation))
         }
 
-        fun handleInvalidInputTypeForAggFun(sourceLocation: SourceLocationMeta, aggFunc: SymbolPrimitive, actualType: StaticType, expectedType: StaticType) {
+        private fun handleInvalidInputTypeForAggFun(sourceLocation: SourceLocationMeta, funcName: String, actualType: StaticType, expectedType: StaticType) {
             problemHandler.handleProblem(
                 Problem(
                     sourceLocation = sourceLocation,
                     details = SemanticProblemDetails.InvalidArgumentTypeForFunction(
-                        functionName = aggFunc.text,
+                        functionName = funcName,
                         expectedType = expectedType,
                         actualType = actualType
                     )
                 )
             )
         }
 
-        private fun computeReturnTypeForAggFunc(aggFunc: SymbolPrimitive, argType: StaticType, sourceLocation: SourceLocationMeta): StaticType {
-            // nested type is not supported by agg functions.
-            // i.e. sum(1,2,[3,4]) is not supported
-            val isNullOrMissingOrNumeric: (StaticType) -> Boolean = {
-                it.isSubTypeOf(StaticType.unionOf(StaticType.MISSING, StaticType.NULL, StaticType.NUMERIC))
+        private fun computeReturnTypeForAggFunc(funcName: String, elementType: StaticType, sourceLocation: SourceLocationMeta): StaticType {
+            if (funcName == "count") {
+                return StaticType.INT
```

Comment:
Why does this now return an unconstrained `INT` rather than an `INT8`? The current evaluator returns always returns a long.

Reply:
In the current implementation, unconstraint integer happens to have the same constraint as a long (`INT8`), https://github.com/partiql/partiql-lang-kotlin/blob/main/lang/src/org/partiql/lang/types/StaticType.kt#L393-L397. 

So returning long does not necessarily mean it must be `INT8`. It can be unlimited `INT` as well. And my preference is to make it return an unlimited `INT`, since the unlimited `INT` is also used elsewhere as the default integer type, e.g. the type of expression `1` is `INT`, instead of `INT8`.